src/Block/AdminPreviewBlockService.php

```diff
@@ -0,0 +1,112 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of the Sonata Project package.
+ *
+ * (c) Thomas Rabaix <thomas.rabaix@sonata-project.org>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sonata\AdminBundle\Block;
+
+use Sonata\AdminBundle\Admin\Pool;
+use Sonata\BlockBundle\Block\BlockContextInterface;
+use Sonata\BlockBundle\Block\Service\AbstractBlockService;
+use Symfony\Bundle\FrameworkBundle\Templating\EngineInterface;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\HttpFoundation\Session\Session;
+use Symfony\Component\OptionsResolver\OptionsResolver;
+
+/**
+ * @author Javier Spagnoletti <phansys@gmail.com>
+ */
+final class AdminPreviewBlockService extends AbstractBlockService
+{
+    /**
+     * @var Pool
+     */
+    private $pool;
+
+    public function __construct(string $name, EngineInterface $templating, Pool $pool)
+    {
+        parent::__construct($name, $templating);
+
+        $this->pool = $pool;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function execute(BlockContextInterface $blockContext, Response $response = null)
```

Comment:
This method is very long, can you please extract some methods

Reply:
Done.